pkgs/by-name/mi/mistral-rs/package.nix

```diff
@@ -178,6 +186,32 @@ rustPlatform.buildRustPackage rec {
     "--skip=util::tests::test_parse_image_url"
   ];
 
+  # postFixup = lib.optionalString cudaSupport (
+  #   let
+  #     patchelfCommand = binaryName: ''
+  #       patchelf \
+  #         --add-rpath ${
+  #           lib.makeLibraryPath (
+  #             with cudaPackages;
+  #             [
+  #               libcublas
+  #               libcurand
+  #             ]
+  #           )
+  #         } \
+  #         $out/bin/${binaryName}
+  #     '';
+  #   in
+  #   ''
+  #     ${patchelfCommand "mistralrs-bench"}
+  #     ${patchelfCommand "mistralrs-server"}
+  #   ''
+  # );
+  runtimeDependencies = lib.optionals cudaSupport [
+    cudaPackages.libcublas
+    cudaPackages.libcurand
+  ];
```

Comment:
```suggestion
```

Try this. I guess it's because `autoPatchelfHook` overrides rpath (`--set-rpath`) instead of prepending/appending (`--add-rpath`) it.

https://github.com/NixOS/nixpkgs/blob/122d20b8ba111b5b6d00e8a90431b4cadce097d6/pkgs/build-support/setup-hooks/auto-patchelf.py#L303-L307

But you already have these two in `buildInputs`, and it seems unneeded to add them to `runtimeDependencies` again, see `torch` as an example:

https://github.com/NixOS/nixpkgs/blob/122d20b8ba111b5b6d00e8a90431b4cadce097d6/pkgs/development/python-modules/torch/bin.nix#L45-L77

Reply:
I guess we have to add something like `extraRuntimeDependencies` then.

Reply:
Ignore me, try this:

https://github.com/NixOS/nixpkgs/blob/122d20b8ba111b5b6d00e8a90431b4cadce097d6/pkgs/development/python-modules/torch/bin.nix#L71-L77

Reply:
Thank you for your suggestions !
I tried removing the
```
runtimeDependencies = lib.optionals cudaSupport [
  cudaPackages.libcublas
  cudaPackages.libcurand
];
```
block but it doesn't work better.
Those libs are loaded at runtime (`dlopen` I guess`) and thus are not seen as required by `patchelf`.

Reply:
> I guess we have to add something like extraRuntimeDependencies to autoPatchelfHook then.

`appendRunpathsArray` exists, used for adding `$ORIGIN` in a few places